#!/usr/bin/env python
PACKAGE = "topic_task"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("randomMin",int_t,0,"the smallest random number", 0,0,100)
gen.add("randomMax",int_t,0,"the biggest random number",10,0,100)

log_enum = gen.enum([ gen.const("info", int_t, 0, "log print flag:INFO"),
                      gen.const("warn", int_t, 1, "log print flag:WARN"),
                      gen.const("error", int_t, 2, "log print flag:ERROR")],
                      "Set Log Level")
gen.add("log_level", int_t, 0, "Set Log Level", 0, 0, 2, edit_method=log_enum)

exit(gen.generate(PACKAGE, "topic_task_node", "dynamic"))
